client/analytics/report/orders/store/actions.js

```diff
@@ -0,0 +1,42 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { dispatch } from '@wordpress/data';
+import apiRequest from '@wordpress/api-request';
+
+export default {
+	setOrders( orders ) {
+		return {
+			type: 'SET_ORDERS',
+			orders,
+		};
+	},
+	updateOrder( order ) {
+		return {
+			type: 'UPDATE_ORDER',
+			order,
+		};
+	},
+	requestUpdateOrder( order ) {
+		return async () => {
+			// Lets be optimistic
+			dispatch( 'wc-admin' ).updateOrder( order );
+			try {
+				const updatedOrder = await apiRequest( {
+					path: '/wc/v3/orders/' + order.id,
+					method: 'PUT',
+					data: order,
+				} );
+
+				dispatch( 'wc-admin' ).updateOrder( updatedOrder );
+			} catch ( error ) {
+				if ( error && error.responseJSON ) {
```

Comment:
I think this is fine for now, but we will want to come up with a standard way to handle these sorts of exceptions.